### PR TITLE
Add logging. Make readyState an enum. Remove recursive call.

### PR DESCRIPTION
**Pull request by drichelson**
Add logging. Make readyState an enum.

### CODE REVIEW

File: src/main/java/com/launchdarkly/eventsource/ReadyState.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.launchdarkly.eventsource;
+
+public enum ReadyState {
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/eventsource/ReadyState.java]: This doesn't need to be public, right? It's an internal concern.

**Comment by jkodumal** [comment]
:+1:

File: build.gradle [hunk 2]
```diff
@@ -21,11 +18,11 @@ allprojects {
 }
 
 dependencies {
-    compile 'com.squareup.okhttp3:okhttp:3.2.0'
+    compile "com.squareup.okhttp3:okhttp:3.2.0"
     compile "org.slf4j:slf4j-api:1.7.7"
-    testRuntime "org.slf4j:slf4j-simple:1.7.7"
+    testRuntime "ch.qos.logback:logback-classic:1.1.3"
```
Comment (drichelson) [inline build.gradle]: made these strings more consistent. Swapped out log4j-simple for logback for more useful log statements in tests.

File: build.gradle [hunk 3]
```diff
@@ -4,13 +4,10 @@ apply plugin: 'signing'
...
 
 repositories {
-    mavenCentral()
     mavenLocal()
-
     // Before LaunchDarkly release artifacts get synced to Maven Central they are here along with snapshots:
-    maven {
-        url "https://oss.sonatype.org/content/groups/public/"
-    }
+    maven { url "https://oss.sonatype.org/content/groups/public/" }
```
Comment (drichelson) [inline build.gradle]: It turns out order matters here.

File: src/main/java/com/launchdarkly/eventsource/EventSource.java [hunk 4]
```diff
@@ -83,43 +85,59 @@ public void close() throws IOException {
   }
 
   private void connect() {
-    Request.Builder builder = new Request.Builder().headers(headers).url(uri.toASCIIString()).get();
-    if (lastEventId != null && !lastEventId.isEmpty()) {
-      builder.addHeader("Last-Event-ID", lastEventId);
-    }
     Response response = null;
     try {
```
Comment (drichelson) [inline src/main/java/com/launchdarkly/eventsource/EventSource.java]: These changes to the connect() method remove the recursive call and also remove our call to handler.onError() when we get an EOFException indicating a disconnect.

**Comment by jkodumal** [comment]
:+1:

**Pull request merged**